### PR TITLE
Feat: skip validating version check

### LINKED ISSUE #4199: Error: addon terraform-alibaba system requirement miss match: deployments.apps "kubevela-vela-core" not found
**Describe the bug**

```
$ vela addon enable terraform-alibaba
Error: addon terraform-alibaba system requirement miss match: deployments.apps "kubevela-vela-core" not found
```

**To Reproduce**

1. install kubevela chart with another release name like `ack-kubevela`
2. enable add-on terraform-alibaba 

**Expected behavior**

**Screenshots**

**KubeVela Version**

**Cluster information**

**Additional context**

The root cause might lie in https://github.com/kubevela/kubevela/blob/a9210487fa5fcf623e4345055aaa6c9def288d80/pkg/addon/addon.go#L1517.

### PR DESCRIPTION
**Pull request by wangyikewxgm**
1. fix bug if specify a helm release name rather than kubevela when install cannot enable addon.
2. add flag skip check validating system version
Signed-off-by: 楚岳 <wangyike.wyk@alibaba-inc.com>

### Description of your changes

Fixes #4199

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: charts/vela-core/templates/kubevela-controller.yaml [hunk 1]
```diff
@@ -122,6 +122,7 @@ metadata:
   name: {{ include "kubevela.fullname" . }}
   namespace: {{ .Release.Namespace }}
   labels:
+    controller.oam.dev/name: application
```
Comment (wonderflow) [inline charts/vela-core/templates/kubevela-controller.yaml:125]:
```suggestion
    controller.oam.dev/name: vela-core
```

You need to check if user upgrade to this version have this label.

Comment (wangyikewxgm) [reply charts/vela-core/templates/kubevela-controller.yaml]: Yes.

File: pkg/addon/addon.go [hunk 2]
```diff
@@ -1513,10 +1521,14 @@ func checkSemVer(actual string, require string) (bool, error) {
...
 
 func fetchVelaCoreImageTag(ctx context.Context, k8sClient client.Client) (string, error) {
-	deploy := &appsv1.Deployment{}
-	if err := k8sClient.Get(ctx, types2.NamespacedName{Namespace: types.DefaultKubeVelaNS, Name: types.KubeVelaControllerDeployment}, deploy); err != nil {
+	deployList := &appsv1.DeploymentList{}
+	if err := k8sClient.List(ctx, deployList, client.MatchingLabels{oam.LabelControllerName: oam.ApplicationControllerName}); err != nil {
 		return "", err
 	}
+	if len(deployList.Items) == 0 {
+		return "", errors.New("System hasn't install vela yet, please install first")
```
Comment (wonderflow) [inline pkg/addon/addon.go:1529]:
```suggestion
		return "", errors.New("can't find a running vela instace running, please install it first")
```

File: charts/vela-core/templates/kubevela-controller.yaml [hunk 3]
```diff
@@ -122,6 +122,7 @@ metadata:
   name: {{ include "kubevela.fullname" . }}
   namespace: {{ .Release.Namespace }}
   labels:
+    controller.oam.dev/name: vela-core
```
Comment (wonderflow) [inline charts/vela-core/templates/kubevela-controller.yaml:125]: according to the helm chart rule, you should add the label here: https://github.com/kubevela/kubevela/blob/master/charts/vela-core/templates/_helpers.tpl#L35-L44

BTW, don't forget add for vela-minimal

Comment (wangyikewxgm) [reply charts/vela-core/templates/kubevela-controller.yaml:125]: > according to the helm chart rule, you should add the label here: https://github.com/kubevela/kubevela/blob/master/charts/vela-core/templates/_helpers.tpl#L35-L44
> 
> BTW, don't forget add for vela-minimal

As far as I Know, labels define in here will added to all types of resource (include configmap,secret) and so on. My point is to only add in controller's deployment.

**Review by wonderflow — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

**Pull request merged**